src/module/Subject/src/Subject/Manager/SubjectManager.php

```diff
@@ -95,10 +95,10 @@ protected function isTrashed(EntityInterface $entity, Collection $collection)
     public function getUnrevisedRevisions(TaxonomyTermInterface $term)
     {
         $entities = $this->entityManager->findAllUnrevised();
-        
+
         // collection for filtered entities by $term
         $filteredEntities = new ArrayCollection();
-        
+
         // find all entities where $term matches (also in parents)
         foreach ($entities as $entity) {
             foreach ($entity->getTaxonomyTerms() as $tempTerm) {
```

Comment:
... so this logic only works for the parents. Hence `$filteredEntities` doesn't include the children